pykafka/producer.py

```diff
@@ -237,7 +428,167 @@ def produce(self, messages):
         :param messages: The messages to produce
         :type messages: Iterable of str or (str, str) tuples
         """
-        # Do partition distribution here. We need to be able to retry producing
-        # only *some* messages when a leader changes. Therefore, we don't want
-        # a random partition distribution changing that on the retry.
-        self._produce(self._partition_messages(messages), 0)
+        super(SynchronousProducer, self).produce(messages)
+        self._wait_all()
```

Comment:
If this will be the only difference between the two producers, couldn't we just have one producer, with a `synchronous=True` kwarg on the `produce` method?